index.js

```diff
@@ -84,6 +98,36 @@ class K8sVMExecutor extends Executor {
                     throw new Error(`Failed to create pod: ${JSON.stringify(resp.body)}`);
                 }
 
+                return resp.body.metadata.name;
+            })
+            .then(podname => new Promise(resolve => setTimeout(resolve, 3000))
+                .then(() => {
+                    const statusOptions = {
+                        uri: `${this.podsUrl}/${podname}/status`,
+                        method: 'GET',
+                        headers: { Authorization: `Bearer ${this.token}` },
+                        strictSSL: false,
+                        maxAttempts: MAXATTEMPTS,
+                        retryDelay: RETRYDELAY,
+                        retryStrategy: podRetryStrategy,
+                        json: true
+                    };
+
+                    return this.breaker.runCommand(statusOptions);
+                }))
+            .then((resp) => {
+                if (resp.statusCode !== 200) {
+                    throw new Error(`Failed to get pod status: ${JSON.stringify(resp.body)}`);
+                }
+
+                console.log(resp);
+
+                const status = resp.body.status.phase.toLowerCase();
+
+                if (status === 'failed' || status === 'unknown') {
```

Comment:
I was looking at documentation for `requestretry`. This won't do what we want. We need to poll status even on `200` success requests. AFAIK https://www.npmjs.com/package/requestretry retries on error. What we are polling is on success `200` requests until status changes to `failed/running`

Reply:
I think that's the default case. If retryStrategy is defined, it should use the retryStrategy:
https://github.com/FGRibreau/node-request-retry/blob/master/index.js#L85